tabular/src/autogluon/tabular/models/abstract/abstract_model.py

```diff
@@ -93,6 +93,10 @@ def __init__(self, path: str, name: str, problem_type: str, eval_metric: Union[s
             self.nondefault_params = list(hyperparameters.keys())[:]  # These are hyperparameters that user has specified.
         self.params_trained = dict()
 
+        #for warm-starting with hyperband scheduler
+        self._prev_model = None #the reference to the current trained model, used as the starting point for incremental training
```

Comment:
This feels unnecessary. Why keep it as a variable in the object instead of a temporary variable in the method?

Reply:
This approach is actually consistent with APIs from CatBoost, Scikit-learn etc. Only warm_start is needed in the method. This means that the class object keeps track of the previous model.

Reply:
Why doesn't just tracking `prev_model` in `._fit` work? The class object does keep track of the previous model already via `self.model`.